packages/@aws-cdk/aws-appsync/lib/schema-intermediate.ts

```diff
@@ -388,3 +386,97 @@ export class UnionType implements IIntermediateType {
     this.definition[options.field?.toString() + 'id'] = options.field;
   }
 }
+
+/**
+ * Properties for configuring an Enum Type
+ *
+ * @experimental
+ */
+export interface EnumTypeOptions {
+  /**
+   * the attributes of this type
+   */
+  readonly definition: string[];
+}
+
+/**
+ * Enum Types are abstract types that includes a set of fields
+ * that represent the strings this type can create.
+ *
+ * @experimental
+ */
+export class EnumType implements IIntermediateType {
+  /**
+   * the name of this type
+   */
+  public readonly name: string;
+  /**
+   * the attributes of this type
+   */
+  public readonly definition: { [key: string]: IField };
+  /**
+   * the authorization modes for this intermediate type
+   */
+  protected modes?: AuthorizationType[];
+
+  public constructor(name: string, options: EnumTypeOptions) {
+    this.name = name;
+    this.definition = {};
+    options.definition.map((fieldName: string) => this.addField({ fieldName }));
+  }
+
+  /**
+   * Create an GraphQL Type representing this Enum Type
+   */
+  public attribute(options?: BaseTypeOptions): GraphqlType {
+    return GraphqlType.intermediate({
+      isList: options?.isList,
+      isRequired: options?.isRequired,
+      isRequiredList: options?.isRequiredList,
+      intermediateType: this,
+    });
+  }
+
+  /**
+   * Method called when the stringifying Intermediate Types for schema generation
+   *
+   * @internal
+   */
+  public _bindToGraphqlApi(api: GraphqlApi): IIntermediateType {
+    this.modes = api.modes;
+    return this;
+  }
+
+  /**
+   * Generate the string of this enum type
+   */
+  public toString(): string {
+    return shapeAddition({
+      prefix: 'enum',
+      name: this.name,
+      fields: Object.keys(this.definition),
+      modes: this.modes,
+    });
+  }
+
+  /**
+   * Add a field to this Enum Type
+   *
+   * To add a field to this Enum Type, you must only configure
+   * addField with the fieldName options.
+   *
+   * @param options the options to add a field
+   */
+  public addField(options: AddFieldOptions): void {
+    if (options.field) {
+      throw new Error('Enum Types do not support IField properties. Use the fieldName option instead of the field option.');
+    }
+    if (!options.fieldName) {
+      throw new Error('When adding a field to an Enum Type, you must configure the fieldName option.');
+    }
+    if (options.fieldName.indexOf(' ') > -1) {
+      throw new Error(`The allowed values of an Enum Type must not have white space. Remove the spaces in "${options.fieldName}"`);
```

Comment:
tighten this message up 
```suggestion
      throw new Error(`Enum Type values cannot have whitespace. Received: "${options.fieldName}"`);
```